# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1227547623 18000
#      Mon Nov 24 12:27:03 2008 -0500
# Node ID 9f34f7636fe0a97b8ff40d6229e8c6c859ca67be
# Parent  39b9f7dfbf4ac3388d6956563cc3aec0812a0169
legend.m: Correct ording of legend labels.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-11-24  Ben Abbott <bpabbott@mac.com>
+
+	* plot/legend.m: Correct ording of legend labels.
+
 2008-11-21  Radek Salac  <salac.r@gmail.com>
 
 	* sparse/cgs.m, sparse/treelayout.m: New functions.
 	* sparse/Makefile.in (SOURCES): Add them here.
 
 2008-11-14  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m (do_tics_1): Support the minorick properties
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -160,21 +160,23 @@ function legend (varargin)
 	    endif
 	endswitch
 	nargs--;
       else
 	varargin = cellstr (arg);
 	nargs = numel (varargin);
       endif
     elseif (iscellstr (arg))
-      varargin = arg;
+      varargin = fliplr (arg);
       nargs = numel (varargin);
     else
       error ("legend: expecting argument to be a character string");
     endif
+  else
+    varargin(1:nargs) = fliplr (varargin(1:nargs));
   endif
 
   if (nargs > 0)
     have_data = false;
     for i = 1:nkids
       if (strcmp (get (kids(k), "type"), "line")
 	  || strcmp (get (kids(k), "type"), "surface")
 	  || strcmp (get (kids(k), "type"), "patch")
@@ -226,16 +228,22 @@ function legend (varargin)
   if (turn_on_legend)
     set (ca, "key", "on");
   endif
 
 endfunction
 
 %!demo
 %! close all;
+%! plot(1:10, 1:10, 1:10, fliplr(1:10));
+%! title("incline is blue and decline is green");
+%! legend({"I'm blue", "I'm green"}, "location", "east")
+
+%!demo
+%! close all;
 %! plot(1:10, 1:10);
 %! title("a very long label can sometimes cause problems");
 %! legend({"hello world"}, "location", "northeastoutside")
 
 %!demo
 %! close all;
 %! labels = {};
 %! for i = 1:5
